ScoutSuite/providers/salesforce/authentication_strategy.py

```diff
@@ -0,0 +1,73 @@
+import requests
+from xml.etree import ElementTree
+from ScoutSuite.providers.base.authentication_strategy import AuthenticationStrategy, AuthenticationException
+
+
+class SalesforceCredentials:
+
+    def __init__(self,
+                 username,
+                 endpoint,
+                 session_id,
+                 metadata_url):
+
+        self.username = username
+        self.endpoint = endpoint
+        self.session_id = session_id
+        self.metadata_url = metadata_url
+
+
+class SalesforceAuthenticationStrategy(AuthenticationStrategy):
+    """
+    Implements authentication for the Openstack provider
+    """
+
+    def authenticate(self,
+                     sf_username,
+                     sf_password,
+                     sf_endpoint,
+                     **kwargs):
+
+        try:
+
+            login_payload = '''<?xml version="1.0" encoding="utf-8" ?>
+                               <env:Envelope xmlns:xsd="http://www.w3.org/2001/XMLSchema"
+                                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                                   xmlns:env="http://schemas.xmlsoap.org/soap/envelope/">
+                                 <env:Body>
+                                   <n1:login xmlns:n1="urn:partner.soap.sforce.com">
+                                     <n1:username>{}</n1:username>
+                                     <n1:password>{}</n1:password>
+                                   </n1:login>
+                                 </env:Body>
+                               </env:Envelope>'''.format(sf_username, sf_password)
+
+            login_headers = {'Content-Type': 'text/xml', 'SOAPAction': 'login'}
+
+            try:
+                # Validate the endpoint URL
+                response = requests.post('https://{}/services/Soap/u/48.0'.format(sf_endpoint),
+                                         data=login_payload, headers=login_headers)
+            except Exception as e:
+                raise AuthenticationException('Login request failed, validate endpoint: {}'.format(e))
+
+            else:
+
+                # TODO "The xml.etree.ElementTree module is not secure against maliciously constructed data.
+                #  If you need to parse untrusted or unauthenticated data see XML vulnerabilities."
```

Comment:
etree is only vulnerable to DoS, so frankly likely not a huge concern. 
In pursuit of best practices however - defusedxml is a drop in replacement.

https://pypi.org/project/defusedxml/#defusedxml-elementtree